### PR TITLE
Replace `NO_FIELDS` by `NoTangent()`

### PR DESCRIPTION
**Pull request by mzgubic**
Follow up to https://github.com/JuliaDiff/ChainRulesCore.jl/pull/358

### CODE REVIEW

**Review by oxinabox — APPROVED** [review]

**Comment by mzgubic** [comment]
Are we ok with this as a non-breaking change?

**Comment by oxinabox** [comment]
The problem with #415  is that it makes any change to CHainRulesCore also breaking for us. so I think we have to say this is breaking.
While we do so maybe we want to also address #415 + delete SpecialFunctions.jl code + delete NaNMath code (noone uses it AFAIK)

File: Project.toml [hunk 1]
```diff
@@ -4,6 +4,7 @@ version = "0.8.0"
 
 [deps]
 ChainRulesCore = "d360d2e6-b24c-11e9-a2a3-2a2ae2dbcce4"
+ChainRulesTestUtils = "cdddcdb0-9152-4a09-a978-84456f9df70a"
```
Comment (oxinabox) [inline Project.toml:7]: seems to have been added by mistake as a diriect dependency

```suggestion
```

**Comment by mzgubic** [comment]
I think we can ignore the SpecialFunctions failure since we will delete those anyway

**Pull request merged**